app/controllers/assessment/grading.rb

```diff
@@ -622,8 +622,8 @@ def load_gradesheet_data
   
     # section filter
     o = params[:section] ? {
-      :conditions => { :assessment_id => id, :users => { :section => @user.section } },
-      :joins => "INNER JOIN users ON submissions.user_id = users.id" # hash doesn't work, Rails = stupid
+      :conditions => { :assessment_id => id, :course_user_data => { :section => @cud.section } },
+      :joins => "INNER JOIN course_user_data ON submissions.course_user_datum_id = course_user_data.id"
```

Comment:
This line isn't even ever used, so you should remove it.  Other than that, if this is tested to work, you can pull it.